File: 프로그래머스/lv2/84512. 모음 사전/모음 사전.py
def solution(word):
    def make_dic(p, depth):
        if depth > 5:
            return
        if p != '':
            data.append(p)
        for x in 'AEIOU':
            make_dic(f'{p}{x}', depth + 1)
    
    answer = 0
    data = []
    make_dic('', 0)
    
    for i in range(len(data)):
        if data[i] == word:
            return i + 1
    
    return answer

File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
class Solution {
    
    public int solution(String word) {
        String v = "AEIOU";
        int[] weights = {781, 156, 31, 6, 1};
        int answer = word.length();

        for (int i = 0; i < word.length(); i++) {
            answer += v.indexOf(word.charAt(i)) * weights[i];
        }

        return answer;
    }
}